sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -1428,6 +1428,47 @@ def explode(self, column, ignore_index):
             preserves_partition_by=preserves,
             requires_partition_by=partitionings.Arbitrary()))
 
+  @frame_base.args_to_kwargs(pd.DataFrame)
+  @frame_base.populate_defaults(pd.DataFrame)
+  def insert(self, value, **kwargs):
+    if isinstance(value, list):
+      raise frame_base.WontImplementMethod(
+          "insert(value=list) is not supported because it joins the input "
+          "list to the deferred DataFrame based on the order of the data.",
+          reason="order-sensitive")
```

Comment:
Is it possible to convert the list into a ConstantExpression?

Reply:
Yeah we could do that, but the list still won't have an index to use as a join key. To match the pandas semantics we'd need to join based on the order of the data.